### PR TITLE
243 bug in removing corner frontiers theyre not always removed

### PR DESCRIPTION
**Pull request by Raa37**
Testing: Ran 1200 simulations to completion

### CODE REVIEW

File: DotBot.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def __init__(self, dotBotId, x, y, floorplan):
         # estimated PDR is number of packets received over number of packets expected
         self.estimatedPdr         = 1      # packets received per second
         # how frequent estimated PDR is sent to orchestrator
-        self.estimatedPdrPeriod   = 10     # in seconds
+        self.estimatedPdrPeriod   = 5     # in seconds
```
Comment (twatteyne) [inline DotBot.py:54]: why the change? I don't disagree, just trying to understand the reasoning to speed things up.

File: SimEngine.py [hunk 2]
```diff
@@ -93,6 +93,19 @@ def run(self):
...
             log.critical(f"Simulation failed with error {traceback.format_exc().splitlines()}")
+
+            # collect completion time
+            self.dataCollector.collect(
+                {
+                    'type':             'KPI',
+                    'timeToCompletion': None,
+                },
+            )
+
```
Comment (twatteyne) [inline SimEngine.py]: remove newline

File: SimEngine.py [hunk 3]
```diff
@@ -93,6 +93,19 @@ def run(self):
...
+            self.dataCollector.flush()
+
+            log.info("Simulation Failed")
```
Comment (twatteyne) [inline SimEngine.py:107]: add a comment `# log`

File: Wireless.py [hunk 4]
```diff
@@ -92,6 +92,7 @@ def __init__(self):
         self.lastPositions   = {}
         self.lastStabilities = {}
 
+
```
Comment (twatteyne) [inline Wireless.py]: remove extra newline (1 is enough)

File: Wireless.py [hunk 5]
```diff
@@ -119,7 +120,10 @@ def transmit(self, sender, frame):
 
             # get pdr between sender and receiver
             pdr            = self._getPDR(sender, relays, receiver)
-            log.debug(f'PDR between {(sender.x, sender.y)} and {(receiver.x, receiver.y)} is {pdr}')
+
+            # # only log pdr when pdr is critically low
```
Comment (twatteyne) [inline Wireless.py]: remove duplicate `#`

File: Wireless.py [hunk 6]
```diff
@@ -119,7 +120,10 @@ def transmit(self, sender, frame):
...
+            if pdr < 0.1:
```
Comment (twatteyne) [inline Wireless.py]: magic value, define 0.1 above

File: Wireless.py [hunk 7]
```diff
@@ -159,6 +163,9 @@ def _getStability(self, sender, receiver):
             (receiver.dotBotId, sender.dotBotId)
         }.intersection(self.lastStabilities.keys())
 
+        # convert set to list and sort, to maintain order for  reproducibility
+        linkInLastStabilities = list(sorted(list(linkInLastStabilities)))
```
Comment (twatteyne) [inline Wireless.py]: is the right-most `list(` needed? It looks like `linkInLastStabilities` is a list.
Also, please detail what this is so we can agree on a name for it

File: configs/razanne.toml [hunk 8]
```diff
@@ -1,8 +1,8 @@
 [atlas]
-    numRobots                = 20
+    numRobots                = 15
     floorplan                = "scattered_obstacles"           # filename in maps
     navigationAlgorithm      = "Atlas"                         # options: Atlas, Ballistic
-    relayAlgorithm           = ["Recovery", "SelfHealing"]     # options: Recovery, Naive, SelfHealing
+    relayAlgorithm           = "Recovery"                      # options: Recovery, SelfHealing, None
```
Comment (twatteyne) [inline configs/razanne.toml]: what does "None" do? No relays? If that is the case, rename to `NoRelays` (`None` is something in Python, it's confusing to use this name)

File: tests/test_computeHeadingSpeedMovementTimeout.py [hunk 9]
```diff
@@ -48,7 +48,7 @@
             'y':        1,
         },
         'out': {
-            'heading':  180,
+            'heading':  270,
```
Comment (twatteyne) [inline tests/test_computeHeadingSpeedMovementTimeout.py:51]: why did this change?

**Review by twatteyne — CHANGES_REQUESTED** [review]
tiny comments

File: DotBot.py (hunk 1 above)
Comment (Raa37) [reply DotBot.py:54]: because with 10 seconds the robots would sometimes lose connectivity before the orchestrator receives a new PDR estimation as there is a large delay between actual PDR and the estimated value the orchestrator has

File: tests/test_computeHeadingSpeedMovementTimeout.py (hunk 9 above)
Comment (Raa37) [reply tests/test_computeHeadingSpeedMovementTimeout.py:51]: Because before we moved DotBot to cell centre always (top left corner + 0.25).
Now we move it to a random position in the cell by adding a random shift instead of the 0.25. 
We also have the option of not adding a shift at all. so the revised values are from moving to top left corner instead of cell centre.

**Review by twatteyne — APPROVED** [review]
